File: 804_Unique_Morse_Code_Words/sol.cpp
class Solution {
public:
    vector<string> alphabet = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
    string process(string word){
        vector<string> tmp;
        for(char& c: word){
            tmp.push_back(alphabet[int(c) - 97]);
        }
        std::string result;
        for (auto const& s : tmp) { result += s; }
        return result;
    }
    
    int uniqueMorseRepresentations(vector<string>& words) {
        transform(words.begin(), words.end(), words.begin(), [this](string& s){
            return process(s);
        });
        std::set<string> s(words.begin(), words.end());
        return s.size();
    }
};

File: 804_Unique_Morse_Code_Words/sol.py
class Solution:
    def __init__(self):
        self.alphabet = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        
    def process(self, word: str):
        return "".join([self.alphabet[ord(s) - 97] for s in word])
    
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        return len(set([self.process(word) for word in words]))
        
